platform/darwin/src/MGLLocationManager.m

```diff
@@ -0,0 +1,104 @@
+#import "MGLLocationManager_Private.h"
+
+@interface MGLCLLocationManager()<CLLocationManagerDelegate>
+
+@property (nonatomic) CLLocationManager *locationManager;
```

Comment:
Why create a default location manager that owns a CLLocationManager? Why not extend CLLocationManager to conform to MGLLocationManager?

Reply:
Te reason for `MGLCLLocationManager` to implement  `MGLLocationManager` is that `MGLMapView` uses the latter for handling authorization/start/stop events and breaks the dependency on `CLLocationManager`.

Reply:
Suppose we instead declare a category on CLLocationManager that conforms to MGLLocationManager. Wouldn’t that accomplish the same thing as this class, except with less indirection?

Reply:
You're right. I changed the implementation accordingly.